src/clusterfuzz/_internal/tests/core/metrics/monitor_test.py

```diff
@@ -157,6 +157,21 @@ def test_cumulative_distribution_metric_fixed(self):
         2,
     ], result.buckets)
 
+  def test_gauge_metric_success(self):
+    """Test gauge metric success."""
+    self.assertIsInstance(
+        monitor.GaugeMetric('g', 'desc', field_spec=None), monitor._GaugeMetric)
+
+  def test_gauge_metric_failure(self):
+    """Test gauge metric failure."""
+    self.mock.check_module_loaded.return_value = False
+    gauge = monitor.GaugeMetric(
+        'g', 'desc', field_spec=[
+            monitor.StringField('name'),
+        ])
+    gauge.set(5)
+    self.assertIsInstance(gauge, monitor._MockMetric)
+
```

Comment:
Can we have a test for the thing that is actually changing in this PR? i.e. the fact that the time series sent to the monitoring service is actually ordered?

Reply:
Added TestFlusherThread test. In the following test we are using 2 metrics , i.e,  BOT_COUNT(Gauge) and HOST_INCONSISTENT_COUNT(Cumulative) in this order and mocking create_time_series method to retrieve 'time_series' list passed to create_time_series in run() api.
Once time_series list is retrieved we are checking whether the start_time of corresponding metrics is in ascending order or not. Based on that test is passing with fix applied and failing if no fix is applied

Reply:
Thanks very much!